providers.tf

```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 5.55.0"
+      version = ">= 6.0.0"
```

Comment:
```suggestion
      version = "< 7.0.0"
```

Reply:
Oh no, please don't do this! It is recommended practice for a reusable module to pin only the _lower_ bound. The _upper_ bound is for the root module to pin. (Your "example" configs are root modules, so this comment applies only to the `providers.tf` suggestion.)

>Reusable modules should constrain only their minimum allowed versions of Terraform and providers, such as `>= 0.12.0`. This helps avoid known incompatibilities, while allowing the user of the module flexibility to upgrade to newer versions of Terraform without altering the module.
>
>Root modules should use a `~>` constraint to set both a lower and upper bound on versions for each provider they depend on.
>https://developer.hashicorp.com/terraform/language/expressions/version-constraints#terraform-core-and-provider-versions

Reply:
Thats fair, I was just trying to protect folks from upgrading unintentionally. But it makes sense that the root modules would pin to the upper bound.